COMPATIBILITY.md

```diff
@@ -0,0 +1,102 @@
+files, their `BUILD` files, their `bzl` files, their `rust_toolchain`
+definitions, their platform definitions, or their build settings.
+
+A backwards compatible release has the property that it only contains backward
+compatible changes.
+
+`rules_rust` follow [SemVer 2.0.0](https://semver.org/). `rules_rust` promise
+that all minor version number and patch version number releases only contain
+backwards compatible changes.
+
+`rules_rust` don't make any guarantees about compatibility between a released
+version and the state of the `rules_rust` at HEAD.
+
+Backwards incompatible changes will have to follow a process in order to be
+released. Users will be given at least one release where the old behavior is
+still present but deprecated to allow smooth migration of their project to the
+new behavior.
+
+## What Bazel versions are supported?
+
+`rules_rust` support the current Bazel LTS at the time of a `rules_rust`
+release.
+
+Support for the current Bazel rolling release is on the best effort basis. If
+the CI build of `rules_rust` against the current Bazel rolling release is green,
+it has to stay green. If the build is already red (because the new Bazel rolling
+release had incompatible changes that broke `rules_rust`), it is acceptable to
+merge PRs leaving the build red as long as the reason for failure remains the
+same. We hope red CI with the current rolling release will be rare.
+
+`rules_rust` don't promise all new `rules_rust` features available to the
+current Bazel rolling release to be be available to the current Bazel LTS
+release (because Bazel compatibility policy doesn't allow us to make that
+promise, and some new features of `rules_rust` require new Bazel features that
+are only available in Bazel rolling releases). `rules_rust` do promise that new
+features available to the current Bazel rolling release will be available to the
+next Bazel LTS release at latest.
+
+Whenever there is a new Bazel LTS release, all releases of `rules_rust` in the
+first 3 months will support both Bazel LTS versions.
+
+## What host platforms are supported?
+
+The only platform subject to backwards compatibility policy is `Linux` on
+`x86_64`, as the only fully supported platform. Process for moving a best effort
+platform to a supported platform is consensus-based.
+
+## What are the stable APIs of `rules_rust`?
+
+`//rust:defs.bzl` is subject to the backwards compatibility policy. That means
+that anything directly accessible from this file is considered stable.
+
+`//rust/private/…` is not subject to the backwards compatibility policy. Content
+of this package is an implementation detail.
+
+`//cargo`, `//util`, `//tools`, `//test`, `//examples`, `//bindgen`, `//proto`
+and any packages not mentioned by this document are by default not subject to
+the backwards compatibility policy.
+
+Experimental build settings are not subject to the backward compatibility
+policy. They should be added to `//rust:experimental.bzl`.
```

Comment:
What is an example of `//rust:experimental.bzl` and `//rust:incompatible/bzl`?

Reply:
Those files will contain `build_settings` (https://docs.bazel.build/versions/main/skylark/config.html) for experimental/incompatible flags. I expect that this file will be potentially loaded from all private bzl files we have (therefore it cannot depend on them; the content of those 2 files and their loaded bzl files has to be standalone).

I expect that people will just duplicate what is already there, and for the first examples we will have a discussion on the PR that introduces them. Do you think we should give concrete examples here in the policy?

Reply:
>  Do you think we should give concrete examples here in the policy?

No, but I think it'd be good to include a template of sorts so whoever ends up needing to add the first one (and I have a feeling I know who that is 😄) knows exactly what the expectation is. I don't think I've seen any other rules implement incompatibility flags so interested to know if there's something I missed that we might be able to refer users to in the description of this PR or something.

Reply:
I'll upload a PR soon.